README.md

```diff
@@ -1,3 +1,40 @@
-# iostream is a golang package to help with streaming coversions
+# iostream
 
-Primary goal for now is converting an io.WriterAt to an io.Writer
+iostream is a golang library meant for converting between streaming interfaces.
+
+The primary goal of iostream for now is converting an io.WriterAt to an io.Writer as long as the WriterAt has a predictable pattern of spawning multiple goroutines that are try to fill out the file from the start. This was primarily built as a way to use the [AWS s3 download manager](https://godoc.org/github.com/aws/aws-sdk-go/service/s3#hdr-Download_Manager) to write to an `io.Writer` instead of an `io.WriterAt` which usually requires either a full file or a full buffer.
```

Comment:
It doesn't say that specifically but if you check out the implementation of the `WriteAtBuffer` that you linked to in your summary comment (https://github.com/aws/aws-sdk-go/blob/v1.25.18/aws/types.go#L162-L171) you can see that it is basically just a simple buffer being grown over time as more data is written to it. Although it doesn't ever move the initial offset or allow flushing of data from the start of the buffer like this iostream implementation does. Therefore it will always eventually grow to the size of file being downloaded.